solr/solrj/src/java/org/apache/solr/client/solrj/impl/PreemptiveBasicAuthClientCustomizer.java

```diff
@@ -66,15 +68,15 @@ public static void setDefaultSolrParams(SolrParams params) {
   }
 
   @Override
-  public void close() throws IOException {}
-
-  @Override
-  public void setup(Http2SolrClient client) {
+  public void setup(SolrClient client) {
+    if (client instanceof Http2SolrClient == false) {
+      return;
+    }
```

Comment:
@gerlowskija , responding to your feedback:

The code here isn't really different than the behavior today -- if you set the sys prop, it only applies to Http2SolrClient (Jetty).  If we start using this mechanism at not only the Http SolrClient level but also CloudSolrClient or whatever else that may or may not apply to this customizer, then it should skip it.  But I get your concern.  Perhaps the property should reflect the usage... so `solr.solrj.http.jetty.customizer` vs `solr.solrj.http.jdk.customizer` (hypothetical).  With this approach, the implementation should throw an exception if the type is wrong.  And the implementation should reflect it as well, like putting "Jetty" into `PreemptiveBasicAuthClientJettyCustomizer`.  LMK what you think; I'd prefer this way.

Reply:
Perhaps it's a matter of taste, but I'd rather have a common interface with one setup method (as I propose here), than either multiple type-specific classes or type-specific methods.  Perhaps adding generics would help slightly, in terms of making an implementation clearly for a specific type.  Yeah.

> My main interest here is around the Java API / signature itself. It feels to me like the type-widening results in a Java API that's easier for a SolrJ user to misuse, and harder for a Solr developer to implement than it was before.

I don't agree/understand this concern.  I'll update this PR soon-ish after another big PR lands.  Perhaps you'll feel better about it.  If you don't; I'm ready to give up and do what you think I should do (which I'm not 100% clear on what that is).